src/Console/Styling/ConsoleOutputThemeInitializer.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Styling;
+
+use Tempest\Container\Container;
+use Tempest\Container\Initializer;
+use Tempest\Container\Singleton;
+
+#[Singleton]
+final readonly class ConsoleOutputThemeInitializer implements Initializer
```

Comment:
Should a theme really be registered in the container? I figure it's just a config value somewhere that's used when constructing the console?

Reply:
I don't remember why I went with that design.

I think you are right - the ConsoleApplication should probably pull the base theme from the ConsoleConfig. 

When it comes to user-defined themes:

We could add a property to ConsoleCommand there to overwrite the default theme. 

If we want to go crazy and drop final from ConsoleCommand, both 1st and 3rd party packages or organizations could create their own styling, which is pretty neat from package author "branding" perspective.

This way they could create a simple wrapper around ConsoleCommand and use their own brand i.e PlatypusCorpCommand

Reply:
> If we want to go crazy and drop final from ConsoleCommand, both 1st and 3rd party packages or organizations could create their own styling, which is pretty neat from package author "branding" perspective.

We don't want to go crazy now, do we? 😛

Sidenote: let me actually share some background info while we're at it: I'm considering to extract the console code into a separate package that people can use as standalone console replacement. I figure it's about time that `symfony/console` gets some proper competition 😛

That won't happen before we merge all these features though, and we shouldn't worry too much about it for now. I'll take care of extracting stuff all at once. I might need to make some changes down the line for things to work in a standalone context.

/sidenote

So regarding personalization, I think registering the output builder within the container is the way to go, and have the initializer take care of setting up branding. To me, that feels like the approach that's most in line with the "Tempest way"